prototype-to-fast-c++.rst

```diff
@@ -0,0 +1,292 @@
+To me, this is nothing short of a revolution in how I do my work. To show you the new workflow I use, I've take an example of a moderately complicated audio algorithm written in python. With this blog, you'll be able to see how simple it is to convert the high-level python/numpy code into a fully deployable, fast, C++ version.
+
+
+An example: time-scaling an audio file.
+================================================================================
+
+A note on time-scaling:
+__________________________________
+
+
+Time scaling in audio signal processing refers to slowing down or speeding up an audio recording while preserving its original pitch (frequency). You may be familiar with playing back a record or a tape at half its normal speed: this both slows down the music but it also changes the pitch: for example a singer's voice become completely unnatural, a female singer might start sounding more like a male. Instead of the music being in a certain key, say A major, it will play in a different key (E major for example, if speeding up by a factor 1.5x. This is undesirable and many techniques have been developed to allow controlling the playback speed while preserving the original "tone" of the music.
+Time-scaling is useful for example if you're trying to figure out a fast series of notes in an improvisation for example, or if you're learning a language and want hear a speaker at a slower speed...
+
+Some time-scaling technique operates in the "frequency domain", i.e. using a Fourier transform to get a view of which frequencies are present in the music at any given time. The one I chose in this blog is a frequency domain technique that's both simple, and achieves reasonable quality for most (but definitely not all) audio tracks and is described in the paper "Improved  Phase  Vocoder
+Time-Scale  Modification  of  Audio" IEEE  TRANSACTIONS  ON SPEECH  AND  AUDIO  PROCESSING, VOL.  7,  NO.  3,  MAY  1999. See for example: Paper_. If you're interested in the general area of audio time-scaling, the wikipedia entry_ is a very good introduction.
+
+The technique uses a Fast Fourier Transform (FFT), pick peaking to locate pure tones in the audio, and complex rotations to adjust the phase of the Fourier transorm prior to reconstructing the audio signal.
+
+Python/numpy code:
+__________________________________
+The code is shown below. Refer to the paper if you're interested in the "theoretical" explanation for what's being done. As you can see, the numpy implementation is simple but not trivial, requiring several steps. Note that the algorithm uses both real and complex numbers: the output of the rfft routine is a complex array, and the rotations are complex numbers. Copy the code below into a time_scaling.py file.
+
+::
+
+    import numpy as np
+
+
+    def hanning(M):
+        if M < 1: return np.array([])
+        if M == 1: return np.ones(1, float)
+        n = np.arange(0, M)

```

Comment:
pythran should be able to optimize this kind of arange, I'll open an issue on that